src/services/storage/handlers/stake.ts

```diff
@@ -29,18 +30,18 @@ const handlers = {
   async Unstaked (event: EventData, { stakeService }: StorageServices): Promise<void> {
     const { user: account, total, token, amount } = event.returnValues
 
-    const stake = await StakeModel.findOne({ where: { token, account } })
+    let stake = await StakeModel.findOne({ where: { token, account } })
 
     if (!stake) {
```

Comment:
Well, that does not really make sense, right? Can you get `Unstaked` event for `token` and `account` that did not call `Staked` before? I would keep the error here.

Reply:
Yea, theoretically it can not. But let's keep it there if you don't mind

Reply:
Hmmm, I don't like this. Because you are making invalid state a valid one, instead allowing us to detect it (as the error will be logged). Yes there is nothing you can do really about it (and in those cases the right thing to do is usually error-out), but we should find out about these problems and investigate.

Out of curiousity I have wrote some tests for this case in to the SC: https://github.com/rsksmart/rif-marketplace-storage/pull/127

And also with this approach you will be getting actually negative total amount...

So I would say, throw an error ;-)

Reply:
I am confused, as I am throwing an error :)

Reply:
Oh, sorry, you are rightfully confused 😅 Not sure why I thought that you are actually creating new entry there 😳
Sorry! My bad 🙈

Reply:
Probably that's because of force push which made before, so it's my bad=)